src/sentry/static/sentry/app/views/stream/stream.jsx

```diff
@@ -119,7 +121,17 @@ const Stream = createReactClass({
       ? nextSearchId
       : nextSearchId !== this.state.searchId;
 
-    if (searchIdChanged || nextProps.location.search !== this.props.location.search) {
+    // We are using qs.parse with location.search since this.props.location.query
+    // returns the same value as nextProps.location.query
+    let currentSearchTerm = qs.parse(this.props.location.search);
+    let nextSearchTerm = qs.parse(nextProps.location.search);
+
+    let searchTermChanged = !isEqual(
+      omit(currentSearchTerm, 'environment'),
+      omit(nextSearchTerm, 'environment')
+    );
+
+    if (searchIdChanged || searchTermChanged) {
```

Comment:
similarly to the other PR, is there a situation where both the environment from the props would change, and the search term or search id would change, and two fetches would occur?

Reply:
It wouldn't happen at the same time, either you're changing environment or performing some other action